inc/namespace.php

```diff
@@ -0,0 +1,314 @@
+	global $wp_styles;
+
+	$err = generate_hash_for_asset( $wp_styles, $handle, $href );
+	if ( is_wp_error( $err ) ) {
+		trigger_error( sprintf( 'Style %s error [%s]: %s', $handle, $err->get_error_code(), $err->get_error_message() ), E_USER_NOTICE );
+	}
+
+	return $html;
+}
+
+/**
+ * Automatically generate hash for a script.
+ *
+ * Hooked into `script_loader_tag` to automatically generate hashes for
+ * scripts on the filesystem.
+ *
+ * @param string $html Stylesheet HTML tag.
+ * @param string $handle Unique handle for the stylesheet.
+ * @param string $href URL for the stylesheet.
+ * @return string Unaltered stylesheet HTML tag.
+ */
+function generate_hash_for_script( string $tag, string $handle, string $src ) : string {
+	global $wp_scripts;
+
+	$err = generate_hash_for_asset( $wp_scripts, $handle, $src );
+	if ( is_wp_error( $err ) ) {
+		trigger_error( sprintf( 'Script %s error [%s]: %s', $handle, $err->get_error_code(), $err->get_error_message() ), E_USER_NOTICE );
+	}
+
+	return $tag;
+}
+
+/**
+ * Automatically generate hash for an asset.
+ *
+ * @param WP_Dependencies $dependencies
+ * @param string $handle
+ * @return WP_Error|null Error if one occurred, or null if successful.
+ */
+function generate_hash_for_asset( WP_Dependencies $dependencies, string $handle ) : ?WP_Error {
+	$asset = $dependencies->query( $handle );
+	if ( ! $asset ) {
+		return new WP_Error(
+			'altis.security.browser.invalid_asset_handle',
+			sprintf(
+				'Invalid asset handle %s',
+				$handle
+			)
+		);
+	}
+
+	// Translate the script back to a path if possible.
+	$src = $asset->src;
+	$site_url = trailingslashit( site_url() );
+	if ( substr( $src, 0, strlen( $site_url ) ) !== $site_url ) {
+		// Not a local asset, skip.
+		return null;
+	}
+
+	$rel_path = substr( $src, strlen( $site_url ) );
+	$query = '';
+	if ( strpos( $rel_path, '?' ) !== false ) {
+		list( $rel_path, $query ) = explode( '?', $rel_path, 2 );
+	}
+
+	if ( path_is_absolute( $rel_path ) || strpos( $rel_path, '../' ) !== false ) {
+		// Invalid relative path.
+		return new WP_Error(
+			'altis.security.browser.invalid_path',
+			sprintf(
+				'Path "%s" for %s is invalid',
+				$src,
+				$handle
+			),
+			compact( 'handle', 'src' )
+		);
+	}
+
+	$root = Altis\ROOT_DIR;
+	if ( substr( $rel_path, 0, 3 ) === 'wp-' ) {
+		// Core asset, use ABSPATH instead.
+		$root = ABSPATH;
+	}
+
+	$actual_path = path_join( $root, $rel_path );
+	if ( ! file_exists( $actual_path ) ) {
+		// Invalid path.
+		return new WP_Error(
+			'altis.security.browser.file_not_exists',
+			sprintf( 'File for %s does not exist', $handle )
+		);
+	}
+
+	// Generate the hash.
+	$hash = generate_hash_for_path( $actual_path, $asset->ver );
+	if ( empty( $hash ) ) {
+		// Couldn't generate a hash.
+		return new WP_Error(
+			'altis.security.browser.could_not_generate_hash',
+			sprintf( 'Could not generate hash for %s', $handle )
+		);
+	}
+
+	if ( ! set_hash_for_asset( $dependencies, $handle, $hash ) ) {
```

Comment:
This is ok but I think it's harder to read, find and debug this step of the process when it's inside the `if` statement. If the function is performing an action and not just checking a status then it should be moved outside and assigned to a variable.